pkg/api/vm/handler.go

```diff
@@ -717,6 +719,48 @@ func (h *vmActionHandler) createVMImages(templateVersion *harvesterv1.VirtualMac
 	return nil
 }
 
+// Since the LH volume creation and replica scheduling are asynchronous
+// Even the CreateVolume csi call success, the replica scheduling may still fail
+// However for the sc with VolumeBindingMode is Immediate
+// we could check annotation for corresponding error if it exists
+func (h *vmActionHandler) checkAttachable(pvc *corev1.PersistentVolumeClaim) error {
+	// Even Harvester specify default StorageClass,
+	// we check the sc existence for robustness
+	if pvc.Spec.StorageClassName == nil {
+		return fmt.Errorf("volme %v with empty StorageClass is not permitted", pvc.Name)
+	}
+
+	sc, err := h.storageClassCache.Get(*pvc.Spec.StorageClassName)
+	if err != nil {
+		return err
+	}
+
+	if sc.VolumeBindingMode == nil {
+		return fmt.Errorf("volme %v not specify VolumeBindingMode is not permitted", pvc.Name)
+	}
+
+	if *sc.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+		return nil
+	}
+
+	if pvc.Status.Phase != corev1.ClaimBound {
+		return fmt.Errorf("volme %v not bound yet", pvc.Name)
+	}
+
+	pv, err := h.pvCache.Get(pvc.Spec.VolumeName)
+	if err != nil {
+		return err
+	}
+
+	//Even the volume is in degrade mode, it still could be attached
+	scheduleErrAnno := pv.Annotations[longhorntypes.PVAnnotationLonghornVolumeSchedulingError]
+	if scheduleErrAnno != "" && scheduleErrAnno != longhorn.ErrorReplicaScheduleSchedulingFailed {
+		return fmt.Errorf("volme %v with error %v", pvc.Name, scheduleErrAnno)
```

Comment:
@LucasSaintarbor Could you help revise the error message? An example is on: https://user-images.githubusercontent.com/19387129/236769070-7024e4c0-55ee-4e3e-bd5d-5a9a166bba1c.png
Thanks!